### ISSUE TITLE
Feature request: automatic testing of documentation examples

### ISSUE THREAD

**Issue by robin-aws** [dafny-lang/dafny#4680]
This is a common need for the Dafny implementation itself but will also be valuable to anyone documenting Dafny projects. After #4678 we will have [three](https://github.com/dafny-lang/dafny/blob/master/docs/check-examples) [different](https://github.com/dafny-lang/libraries/blob/master/Scripts/check-examples) [copies](https://github.com/dafny-lang/dafny/blob/standard-library-support-verification-only/Source/DafnyStandardLibraries/scripts/check-examples) of the `check-examples` bash script, each a bit different, and it would be great to promote something like it to a proper Dafny ecosystem tool. We probably want the implementation to be in C# along with the rest of the Dafny implementation.

I think the ideal context will be code snippets in Dafny docstrings, since docstrings live inside a module and hence provide the necessary resolution context. But supporting separate Markdown files would be nice to have as well.
